### PR TITLE
build config name contains hash fix

### PR DESCRIPTION
**Pull request by rcerven**

### CODE REVIEW

**Comment by twaugh** [comment]
Will you add an assert statement to verify the name is in the format OpenShift will accept?

**Comment by rcerven** [comment]
I'll do it in that another PR, which will actually resolve problem, so I'll make unit tests for that as well

**Comment by vrutkovs** [comment]
Not sure if removing base64 encoding is a good idea, as we might have hash collision. The alternative way is to `.lower()` the result instead.

Lets see the test results for that

**Comment by twaugh** [comment]
I'd rather not mess around with the output of base64. If we need more bits, let's use more characters for the hexdigest.

**Comment by rcerven** [comment]
I have tested it for all our repo/branches and all sha were unique

File: tests/test_utils.py [hunk 1]
```diff
@@ -104,6 +104,20 @@ def test_make_name_from_git_collide():
...
 
+SHA_INPUT_FILE = 'tests/input_for_sha.txt'
+
+def test_make_name_from_git_all_from_file():
+
+    all_sha = set()
+    with open(SHA_INPUT_FILE) as f:
+        lines =  f.read().splitlines()
+
+    for line in lines:
```
Comment (twaugh) [inline tests/test_utils.py]: Could use:
```
with open(SHA_INPUT_FILE) as f:
    for line in f:
```

Comment (rcerven) [inline tests/test_utils.py]: but I need line count also, so I can either count lines in that loop or leave it as it is

Comment (twaugh) [inline tests/test_utils.py]: Ah yes. Let's leave it as-is.

Comment (lcarva) [inline tests/test_utils.py]: BTW, `enumerate()` is a great way to achieve that.

```python
index = -1
for index, line in enumerate(f):
  ...
number_of_lines = index + 1
```

**Pull request merged**